test/libsolidity/smtCheckerTests/operators/ternary_operator_4.sol

```diff
@@ -0,0 +1,30 @@
+abstract contract D {
+	function d() external virtual returns (uint);
+}
+
+contract C {
+
+  D d;
+  uint v;
+  bool guard = true;
+
+  function inc() public {
+	++v;
+  }
+
+  function dec() public {
+    if (guard) return;
+    --v;
+	guard = true;
```

Comment:
I find this way of using this `guard` a bit weird. Maybe it makes sense to change this and the other test to use a "classic" lock modifier instead? I think the results should be the same.

Reply:
I can try, but I will have to figure out how to do that.
Here I only want to allow a single call to `dec` from external function, in the previous example I want to allow any number of calls.

Reply:
The usual mutex modifier pattern does not mach the semantics required in these tests.